tests/functional/botocore/configured_endpoint_urls/test_configured_endpoint_url.py

```diff
@@ -0,0 +1,249 @@
+        )
+
+
+def get_create_client_args(test_case_client_config):
+    create_client_args = {}
+
+    if 'endpoint_url' in test_case_client_config:
+        create_client_args['endpoint_url'] = test_case_client_config[
+            'endpoint_url'
+        ]
+
+    if 'ignore_configured_endpoint_urls' in test_case_client_config:
+        create_client_args['config'] = Config(
+            ignore_configured_endpoint_urls=test_case_client_config[
+                'ignore_configured_endpoint_urls'
+            ]
+        )
+
+    return create_client_args
+
+
+def get_config_file_contents(profile_name, test_suite):
+    profile = test_suite['profiles'][profile_name]
+
+    profile_str = dict_to_ini_section(
+        profile,
+        section_header=f"profile {profile_name}",
+    )
+
+    services_section_name = profile.get('services', None)
+
+    if services_section_name is None:
+        return profile_str
+
+    services_section = test_suite['services'][services_section_name]
+
+    service_section_str = dict_to_ini_section(
+        services_section,
+        section_header=f'services {services_section_name}',
+    )
+
+    return profile_str + service_section_str
+
+
+@pytest.fixture
+def client_creator(tmp_path):
+    tmp_config_file_path = tmp_path / 'config'
+    environ = {'AWS_CONFIG_FILE': str(tmp_config_file_path)}
+
+    def _do_create_client(
+        service,
+        profile,
+        client_args=None,
+        config_file_contents=None,
+        environment=None,
+    ):
+        environ.update(environment)
+        with open(tmp_config_file_path, 'w') as f:
+            f.write(config_file_contents)
+            f.flush()
+
+        return botocore.session.Session(profile=profile).create_client(
+            service, **client_args
+        )
+
+    with mock.patch('os.environ', environ):
+        yield _do_create_client
+
+
+def _normalize_endpoint(url):
+    split_endpoint = urlsplit(url)
+    actual_endpoint = f"{split_endpoint.scheme}://{split_endpoint.netloc}"
+    return actual_endpoint
+
+
+def assert_client_endpoint_url(client, expected_endpoint_url):
+    assert client.meta.endpoint_url == expected_endpoint_url
+
+
+def assert_endpoint_url_used_for_operation(
+    client, expected_endpoint_url, operation, params
+):
+    http_stubber = ClientHTTPStubber(client)
+    http_stubber.start()
+    http_stubber.add_response()
+
+    # Call an operation on the client
+    getattr(client, operation)(**params)
+
+    assert (
+        _normalize_endpoint(http_stubber.requests[0].url)
+        == expected_endpoint_url
+    )
+
+
+def _known_service_names_and_models():
+    my_session = botocore.session.get_session()
+    loader = my_session.get_component('data_loader')
+    available_services = loader.list_available_services('service-2')
+
+    result = []
+    for service_name in available_services:
+        model = my_session.get_service_model(service_name)
+        result.append((model.service_name, model))
```

Comment:
I think we are running into memory issues in the Windows jobs. Specifically, I think it is because we load all of the models as part of parametrization and that lingers in memory until the test execution finish. This could result in quite a bit of memory being used depending on how many copies of all of the models we have in memory from test collection. I think we should change this test to parametrize off of service id's. I tried it on my fork and that seemed to do the trick in getting the Windows CI passing: https://github.com/kyleknap/aws-cli/commit/ee21da40fdb92d22325f0f716e72f36569e77812